pkg/kubelet/cm/cpumanager/topology_hints.go

```diff
@@ -21,26 +21,101 @@ import (
 	"k8s.io/klog"
 
 	"k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/topology"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
 	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/socketmask"
 )
 
 func (m *manager) GetTopologyHints(pod v1.Pod, container v1.Container) []topologymanager.TopologyHint {
+	// The 'none' policy does not generate topology hints.
+	if m.policy.Name() == string(PolicyNone) {
+		return nil
+	}
+
 	var cpuHints []topologymanager.TopologyHint
 	if requestedObj, ok := container.Resources.Requests[v1.ResourceCPU]; ok {
 		// Get a count of how many CPUs have been requested
 		requested := int(requestedObj.Value())
 		klog.Infof("[cpumanager] Guaranteed CPUs detected: %v", requested)
 
-		// Discover topology in order to establish the number
-		// of available CPUs per socket.
-		_, err := topology.Discover(m.machineInfo)
-		if err != nil {
-			klog.Infof("[cpu manager] error discovering topology")
-			return nil
-		}
+		// Get a list of available CPUs.
+		// For now, this implementation assumes the 'static' CPUManager policy.
+		// TODO: Generalize this so that its applicable to future CPUManager polices.
+		available := m.policy.(*staticPolicy).assignableCPUs(m.state)
 
-		// TODO: Fill in cpuHints with proper TopologyHints
+		// Generate hints using the list of available devices and the number of
+		// CPUs being requested.
+		cpuHints = m.generateCPUTopologyHints(available, requested)
 	}
+
 	klog.Infof("[cpumanager] Topology Hints for pod: %v", cpuHints)
 	return cpuHints
 }
+
+func (m *manager) iterateCPUCombinations(cpus cpuset.CPUSet, amount int, callback func(cpuset.CPUSet)) {
```

Comment:
This could have a state explosion.

Imagine, for example, a 100 CPU system, iterating all possible combinations of requesting 50 CPUs.
That's 1.0089134e+29 combinations we would have to walk through.

We can probably do something smarter here...